pkg/plugin/framework/action_resolver.go

```diff
@@ -0,0 +1,278 @@
+	return recv.BackupItemAction
+}
+
+func (recv BackupItemResolvedAction) GetBackupItemAction() velero.BackupItemAction {
+	return recv.BackupItemAction
+}
+
+func NewBackupItemActionResolver(actions []velero.BackupItemAction) BackupItemActionResolver {
+	return BackupItemActionResolver{
+		actions: actions,
+	}
+}
+
+func NewRestoreItemActionResolver(actions []velero.RestoreItemAction) RestoreItemActionResolver {
+	return RestoreItemActionResolver{
+		actions: actions,
+	}
+}
+
+func NewDeleteItemActionResolver(actions []velero.DeleteItemAction) DeleteItemActionResolver {
+	return DeleteItemActionResolver{
+		actions: actions,
+	}
+}
+
+func NewItemSnapshotterResolver(actions []isv1.ItemSnapshotter) ItemSnapshotterResolver {
+	return ItemSnapshotterResolver{
+		actions: actions,
+	}
+}
+
+type ActionResolver interface {
+	ResolveAction(helper discovery.Helper, action velero.Applicable) (ResolvedAction, error)
+}
+
+type BackupItemActionResolver struct {
+	actions []velero.BackupItemAction
+}
+
+func (recv BackupItemActionResolver) ResolveActions(helper discovery.Helper) ([]BackupItemResolvedAction, error) {
+	var resolved []BackupItemResolvedAction
+	for _, action := range recv.actions {
+		resources, namespaces, selector, err := resolveAction(helper, action)
+		if err != nil {
+			return nil, err
+		}
+		res := BackupItemResolvedAction{
+			BackupItemAction: action,
+			resolvedAction: resolvedAction{
+				ResourceIncludesExcludes:  resources,
+				NamespaceIncludesExcludes: namespaces,
+				Selector:                  selector,
+			},
+		}
+		resolved = append(resolved, res)
+	}
+	return resolved, nil
+}
+
+type RestoreItemResolvedAction struct {
+	velero.RestoreItemAction
+	resolvedAction
+}
+
+func (recv RestoreItemResolvedAction) GetApplicable() velero.Applicable {
+	return recv.RestoreItemAction
+}
+
+func (recv RestoreItemResolvedAction) GetRestoreItemAction() velero.RestoreItemAction {
+	return recv.RestoreItemAction
+}
+
+type RestoreItemActionResolver struct {
+	actions []velero.RestoreItemAction
+}
+
+func (recv RestoreItemActionResolver) ResolveActions(helper discovery.Helper) ([]RestoreItemResolvedAction, error) {
+	var resolved []RestoreItemResolvedAction
+	for _, action := range recv.actions {
+		resources, namespaces, selector, err := resolveAction(helper, action)
+		if err != nil {
+			return nil, err
+		}
+		res := RestoreItemResolvedAction{
+			RestoreItemAction: action,
+			resolvedAction: resolvedAction{
+				ResourceIncludesExcludes:  resources,
+				NamespaceIncludesExcludes: namespaces,
+				Selector:                  selector,
+			},
+		}
+		resolved = append(resolved, res)
+	}
+	return resolved, nil
+}
+
+type DeleteItemResolvedAction struct {
+	velero.DeleteItemAction
+	resolvedAction
+}
+
+func (recv DeleteItemResolvedAction) GetApplicable() velero.Applicable {
+	return recv.DeleteItemAction
+}
+
+func (recv DeleteItemResolvedAction) GetRestoreItemAction() velero.DeleteItemAction {
```

Comment:
This is not needed and the func name is incorrect.

Reply:
I thought I'd taken all these out before, thanks